clicommand/agent_start.go

```diff
@@ -835,6 +835,11 @@ var AgentStartCommand = cli.Command{
 			}
 		}
 
+		// default signal grace period to slightly less than the cancel grace period
+		if cfg.SignalGracePeriodSeconds == -1 {
+			cfg.SignalGracePeriodSeconds = cfg.CancelGracePeriod - 1
+		}
```

Comment:
Is it perhaps worth generalising this? e.g.

```suggestion
		if cfg.SignalGracePeriodSeconds < 0 {
			cfg.SignalGracePeriodSeconds = cfg.CancelGracePeriod + cfg.SignalGracePeriodSeconds
		}
```

Reply:
I guess? I was just using `-1` as "unset" in true awful linux fashion but I don't have strong opinions